src/rosdep2/platforms/gentoo.py

```diff
@@ -54,10 +58,21 @@ def register_installers(context):
     context.set_installer(PORTAGE_INSTALLER, PortageInstaller())
 
 def register_platforms(context):
+    register_funtoo(context)
+    register_gentoo(context)
+
+def register_gentoo(context):
     context.add_os_installer_key(OS_GENTOO, PORTAGE_INSTALLER)
     context.add_os_installer_key(OS_GENTOO, SOURCE_INSTALLER)
     context.set_default_os_installer_key(OS_GENTOO, PORTAGE_INSTALLER)
 
+def register_funtoo(context):
+    # Funtoo is an alias for Gentoo.
+    (os_name, os_version) = context.get_os_name_and_version()
+    if os_name == OS_FUNTOO and not context.override:
+        print("rosdep detected OS: [%s], aliasing it to [%s]" % (OS_FUNTOO, OS_GENTOO), file=sys.stderr)
+        context.set_os_override(OS_GENTOO, context.os_detect.get_codename())
```

Comment:
Ah so it seems the codename is not defined for gentoo/funtoo and always returns the empty string `''`. Try `get_version()` instead of `get_codename()` here for the override.